plugin/trino-vertica/src/main/java/io/trino/plugin/vertica/VerticaClient.java

```diff
@@ -0,0 +1,248 @@
+                }
+                if (precision > Decimals.MAX_PRECISION) {
+                    break;
+                }
+                return Optional.of(decimalColumnMapping(createDecimalType(precision, max(decimalDigits, 0))));
+            case Types.CHAR:
+                return Optional.of(charColumnMapping(createCharType(typeHandle.getRequiredColumnSize()), false));
+            case Types.VARCHAR:
+            case Types.LONGVARCHAR:
+                return Optional.of(varcharColumnMapping(createVarcharType(typeHandle.getRequiredColumnSize()), false));
+            case Types.DATE:
+                return Optional.of(dateColumnMapping());
+        }
+        if (getUnsupportedTypeHandling(session) == CONVERT_TO_VARCHAR) {
+            return mapToUnboundedVarchar(typeHandle);
+        }
+
+        return Optional.empty();
+    }
+
+    @Override
+    public WriteMapping toWriteMapping(ConnectorSession session, Type type)
+    {
+        if (type == BOOLEAN) {
+            return WriteMapping.booleanMapping("boolean", booleanWriteFunction());
+        }
+        if (type == SMALLINT) {
+            return WriteMapping.longMapping("smallint", smallintWriteFunction());
+        }
+        if (type == TINYINT) {
+            return WriteMapping.longMapping("tinyint", tinyintWriteFunction());
+        }
+        if (type == INTEGER) {
+            return WriteMapping.longMapping("integer", integerWriteFunction());
+        }
+        if (type == BIGINT) {
+            return WriteMapping.longMapping("bigint", bigintWriteFunction());
+        }
+        if (type == REAL) {
+            return WriteMapping.longMapping("real", realWriteFunction());
+        }
+        if (type == DOUBLE) {
+            return WriteMapping.doubleMapping("double precision", doubleWriteFunction());
+        }
+        if (type instanceof DecimalType) {
+            DecimalType decimalType = (DecimalType) type;
+            String dataType = format("decimal(%s, %s)", decimalType.getPrecision(), decimalType.getScale());
+            if (decimalType.isShort()) {
+                return WriteMapping.longMapping(dataType, shortDecimalWriteFunction(decimalType));
+            }
+            return WriteMapping.sliceMapping(dataType, longDecimalWriteFunction(decimalType));
+        }
+        if (type instanceof CharType) {
+            int length = ((CharType) type).getLength();
+            checkArgument(length <= 65000, "Char length is greater than 65,000");
+            return WriteMapping.sliceMapping("char(" + length + ")", charWriteFunction());
+        }
+        if (type instanceof VarcharType) {
+            VarcharType varcharType = (VarcharType) type;
+            String dataType;
+            if (varcharType.isUnbounded()) {
+                dataType = "long varchar";
+            }
+            else {
+                checkArgument(varcharType.getBoundedLength() <= 32_000_000, "Varchar length is greater than 32,000,000");
```

Comment:
nit: The error message can be more explicit to mention that it's a Vertica limitation. (which also makes the code more clear in my opinion).